app/javascript/react/components/MemesFormContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react'
+import TextField from './TextField'
+
+class MemesFormContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state={
+      title: '',
+      imageUrl: '',
+      description: ''
+
+    }
+    this.handleOnChange = this.handleOnChange.bind(this)
+    this.handleOnSubmit = this.handleOnSubmit.bind(this)
+  }
+
+  handleOnChange(event) {
+
+    let newData = event.target.value
+    this.setState({[event.target.name] : newData})
+  }
+
+  handleOnSubmit(event){
+    event.preventDefault()
+    let memePayload={
+      title: this.state.title,
+      imageUrl: this.state.imageUrl,
+      description: this.state.description
+    };
+    
+    fetch('/api/v1/memes', {
+      credentials: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify(memePayload),
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      }
+    })
+    .then(response => {
+      if (response.ok) {
+        return response;
+      } else {
+        let errorMessage = `${response.status}(${response.statusText})` ,
+        error = new Error(errorMessage);
+        throw(error);
+      }
+    })
+    .then(response => response.json())
+    .then(body => {
+    })
+    .catch(error => console.error(`Error in fetch: ${error.message}`));
+    this.clearForm()
```

Comment:
Ironically, you could very well put an extra line above #53 to make it more readable 😁